test/logflare_web/controllers/api/endpoint_controller_test.exs

```diff
@@ -112,15 +112,15 @@ defmodule LogflareWeb.Api.EndpointControllerTest do
     } do
       name = TestUtils.random_string()
 
-      conn
-      |> login_user(user)
-      |> delete("/api/endpoints/#{endpoint.token}", %{name: name})
-      |> response(204)
-
-      conn
-      |> login_user(user)
-      |> get("/api/endpoints/#{endpoint.token}")
-      |> response(404)
+      assert conn
+             |> login_user(user)
```

Comment:
can move `|> login_user(user)` to setup block

Reply:
or can use a setup function for :valid_user/:invalid_user logins

Reply:
honestly don't seem much benefit, this way the tests are more descriptive of the flow they are executing for a given test

Reply:
sure, just seems like it is being repeated for each test